test/jest/fixtures.js

```diff
@@ -0,0 +1,134 @@
+    const server = await TestServer.create(assetsPath, port);
+    server.enableHTTPCache(cachedPath);
+    server.PORT = port;
+    server.PREFIX = `http://localhost:${port}`;
+    server.CROSS_PROCESS_PREFIX = `http://127.0.0.1:${port}`;
+    server.EMPTY_PAGE = `http://localhost:${port}/empty.html`;
+  
+    const httpsPort = port + 1;
+    httpsServer = await TestServer.createHTTPS(assetsPath, httpsPort);
+    httpsServer.enableHTTPCache(cachedPath);
+    httpsServer.PORT = httpsPort;
+    httpsServer.PREFIX = `https://localhost:${httpsPort}`;
+    httpsServer.CROSS_PROCESS_PREFIX = `https://127.0.0.1:${httpsPort}`;
+    httpsServer.EMPTY_PAGE = `https://localhost:${httpsPort}/empty.html`;
+  
+    await test(server);
+
+    await Promise.all([
+      server.stop(),
+      httpsServer.stop(),
+    ]);  
+  });
+
+  global.registerWorkerFixture('playwright', async({}, test) => {
+    if (process.env.PWCHANNEL) {
+      setUseApiName(false);
+      const connection = new Connection();
+      let toImpl;
+      if (process.env.PWCHANNEL === 'wire') {
+        this.spawnedProcess = childProcess.fork(path.join(__dirname, '..', 'lib', 'rpc', 'server'), [], {
+          stdio: 'pipe',
+          detached: process.platform !== 'win32',
+        });
+        this.spawnedProcess.once('exit', (exitCode, signal) => {
+          this.spawnedProcess = undefined;
+          if (!this.expectExit)
+            throw new Error(`Server closed with exitCode=${exitCode} signal=${signal}`);
+        });
+        process.on('exit', () => this._killProcess());
+        const transport = new Transport(this.spawnedProcess.stdin, this.spawnedProcess.stdout);
+        connection.onmessage = message => transport.send(JSON.stringify(message));
+        transport.onmessage = message => connection.dispatch(JSON.parse(message));
+      } else {
+        const dispatcherConnection = new DispatcherConnection();
+        dispatcherConnection.onmessage = async message => {
+          setImmediate(() => connection.dispatch(message));
+        };
+        connection.onmessage = async message => {
+          const result = await dispatcherConnection.dispatch(message);
+          await new Promise(f => setImmediate(f));
+          return result;
+        };
+        new PlaywrightDispatcher(dispatcherConnection.rootDispatcher(), this._playwright);
+        toImpl = x => dispatcherConnection._dispatchers.get(x._guid)._object;
+      }
+      const playwright = await connection.waitForObjectWithKnownName('playwright');
+      playwright.toImpl = toImpl;
+      await test(playwright);
+    } else {
+      playwright.toImpl = x => x;
+      await test(playwright);
+    }
+  });
+
+  global.registerFixture('toImpl', async ({playwright}, test) => {
+    await test(playwright.toImpl);
+  });
+
+  global.registerWorkerFixture('browserType', async ({playwright}, test) => {
+    await test(playwright[process.env.BROWSER || 'chromium']);
```

Comment:
I think this breaks headful/travis/circle/appveyor that use 'npm run test' meaning 'run tests for all browsers'.

Reply:
I still don't think this is a real scenario. Let's first tackle the per-browser run.